docs: update development guides (#271)
localstack/docs: content/en/developer-guide/parity-testing.md

weight: 7
description: >
  How to run and write Parity tests to verify and improve the correctness of LocalStack compared to AWS.
---

Parity tests are a special form of integration tests that should verify and improve the correctness of LocalStack compared to AWS.

Initially, a parity test is executed against AWS - the test marks certain responses that should be collected and verified for correctness. The result of the initial execution is a **snapshot.json** file that contains the recorded responses.

During a “normal” test execution, the test runs against LocalStack and compares the responses with the recorded content.

## How to write Parity tests

1.  Make sure that the test works against AWS.
    
2.  Add the `snapshot` fixture to your test and identify which responses you want to collect and compare against LocalStack
    
	1.  `snapshot.match(”identifier”, result)` indicates a result that will be recorded and stored in a file with the name `<testfile-name>.snapshot.json`
    
	    The **identifier** can be freely selected, but ideally should give a hint on what is recorded - so typically the name of the function. The **result** is expected to be a `dict`.

> 	2. Run the test against AWS: use the parameter `--snapshot-update` (_**update**_: we now also support the ENV `SNAPSHOT_UPDATE=1`) and set the environment variable as `TEST_TARGET=AWS_CLOUD`.

Reviewer: ```suggestion
	2. Run the test against AWS: use the parameter `--snapshot-update` (or the environment variable `SNAPSHOT_UPDATE=1`) and set the environment variable as `TEST_TARGET=AWS_CLOUD`.
```
